Documentation/pre-built-detection.md

```diff
@@ -0,0 +1,147 @@
+# Pre-built detection guide
+
+This is a detailed guide on how to approach enabling pre-built detection for a source-buildable repository.
+It is primarily intended for repository maintainers in the `dotnet` organization.
+
+## Pre-requisites
+
+This guide places several assumptions on the repository and the maintainer in charge of enabling pre-built detection:
+
+  - the repository in question is source-buildable;
+  - the person performing the task is aware of basic source-build concepts;
+
+## Table of content
+  - [Pre-builts](#pre-builts)
+  - [Enabling mandatory pre-built detection](#enabling-mandatory-pre-built-detection)
+  - [Elimitating pre-builts](#eliminating-pre-builts)
+  - [FAQ](#faq)
+  - [Contacts](#contacts)
+
+# Pre-builts
+
+_Source-build_ is a process of building a given product on a single machine with no internet access.
+As such, all of the upstream packages must be part of the current source-build, since their _pre-built_ versions cannot be retrieved from, for example, `nuget.org` or some internal feed.
+
+By definition, _pre-builts_ are dependencies that a repo has on binary files that are not build from source, where _build from source_ points to any package produced during the _current source-build_.
+In lament terms, this means that packages from `nuget.org`, Microsoft builds or other unrelated source-builds cannot be used for source-building a given repository.
+
+Several different methods are provided to repositories to be able to handle pre-builts for their individual builds while being source-buildable.
+These methods are expanded on in the [Eliminating pre-builts](#eliminating-pre-builts) section of the guide.
+
+# Enabling mandatory pre-built detection
+
+## Motivation
+
+WIP
+
+## Verifying that the repository is source-buildable (?)
+
+WIP
+
+## Identifying used pre-builts
+
+By default, when source-build logic was first introduced to the repository, all pre-builts were allowed.
+This was accomplished by instructing pre-built detection logic to not fail the build for any found pre-builts via a dedicated glob rule.
+The rule itself can be found in the `./eng/SourceBuildPrebuildBaseline.xml` file.
+
+Nevertheless, pre-built detection compiles a report of pre-built usage regardless of any rules set.
```

Comment:
I think it would be useful to provide a brief high level overview of how the prebuild detection works.

Reply:
Agree, added